### PR TITLE
Azure Arc on Azure VMs scenario

### PR DESCRIPTION
**Pull request by lanicolas**

### CODE REVIEW

File: azure_arc_servers_jumpstart/azure/arm_template/azurevm_linux.json [hunk 1]
```diff
@@ -0,0 +1,261 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmName": {
+      "type": "string",
+      "defaultValue": "arc-linux-demo",
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/azure/arm_template/azurevm_linux.json:7]:
```suggestion
      "defaultValue": "Arc-Linux-Demo",
```

File: azure_arc_servers_jumpstart/azure/arm_template/azurevm_linux.json [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+      ],
+      "metadata": {
+        "description": "The Ubuntu version for the VM. This will pick a fully patched image of this given Ubuntu version."
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/azure/arm_template/azurevm_linux.json:59]:
```suggestion
        "description": "Azure region location for all resources."
```

File: azure_arc_servers_jumpstart/azure/arm_template/azurevm_linux.json [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+      }
+    },
+    "VmSize": {
+      "type": "string",
+      "defaultValue": "Standard_B2s",
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/azure/arm_template/azurevm_linux.json:64]:
```suggestion
      "defaultValue": "Standard_B2ms",
```

File: azure_arc_servers_jumpstart/azure/arm_template/azurevm_linux.json [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+      "metadata": {
+        "description": "The size of the VM"
+      }
+    },
+    "virtualNetworkName": {
+      "type": "string",
+      "defaultValue": "vNet",
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/azure/arm_template/azurevm_linux.json:71]:
```suggestion
      "defaultValue": "Azure-Arc-Demo-VNET",
```

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    - **adminUsername:** a username for Admin access to the Linux OS
+    - **adminPublicKey:** the public key for SSH access
+    - **dnsLabelPrefix:** a DNS prefix for the VM 
+    - **vmName:** a custom name for the Azure VM
+
+* To deploy the ARM template, navigate to the local cloned deployment folder and run the below command:
+
+    ```console
+    az deployment group create --resource-group <Name of the Azure Resource Group> --name <The name of this deployment> --template-uri https://raw.githubusercontent.com/microsoft/azure_arc/master/azure_arc_servers_jumpstart/azure/arm_template/azurevm_linux.json --parameters <The *azurevm_linux.parameters.json* parameters file location>
+    ```
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/azure_arm_linux.md:55]: convert to multiple lines command

File: azure_arc_servers_jumpstart/azure/arm_template/azurevm_windows.json [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    },
+    "vmName": {
+      "type": "string",
+      "defaultValue": "arc-azurewin-demo",
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/azure/arm_template/azurevm_windows.json:78]:
```suggestion
      "defaultValue": "Arc-Win-Demo",
```
arc-azurewin-demo is longer then 15 chars

File: README.md [hunk 7]
```diff
@@ -44,6 +44,17 @@ The following examples can be used to connect existing Windows or Linux servers
 
 * [Connect an existing Windows machine to Azure Arc](azure_arc_servers_jumpstart/docs/onboard_server_win.md)
 
+#### Azure 
+
+Azure Arc connected machine agent cannot be deployed on Azure VMs as they are already part of Azure Resource Manager. However, for demo purposes, you can use Azure VMs to host an Azure VM and onboard it to Azure Arc. This will allow you to simulate "on-premises" servers on Azure VMs.
```
Comment (likamrat) [inline README.md:49]:
```suggestion
Azure VMs are leveraging the [Azure Instance Metadata Service (IMDS)](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service) by default. 

By projecting an Azure VM as an Azure Arc enabled server, a "conflict" is created which will not all the Azure Arc server resources to be represented as one when the IMDS is being used and instead, the Azure Arc server will still "act" as a native Azure VM. 

However, for demo purposes, the below guides will allow you to use and onboard Azure VMs to Azure Arc and by that, you will be able to simulate a server which is deployed outside of Azure (i.e "on-premises" or in other cloud platforms)
```

File: README.md [hunk 8]
```diff
@@ -44,6 +44,17 @@ The following examples can be used to connect existing Windows or Linux servers
...
+
+Azure Arc connected machine agent cannot be deployed on Azure VMs as they are already part of Azure Resource Manager. However, for demo purposes, you can use Azure VMs to host an Azure VM and onboard it to Azure Arc. This will allow you to simulate "on-premises" servers on Azure VMs.  
+
+* [Deploy an Ubuntu Azure VM and connect it to Azure Arc using ARM](azure_arc_servers_jumpstart/docs/azure_arm_windows.md)
+
+* [Deploy a Windows Azure VM and connect it to Azure Arc using ARM](azure_arc_servers_jumpstart/docs/azure_arm_linux.md)
+
+    **Note:** Use only for demo purposes
+
+
```
Comment (likamrat) [inline README.md:57]:
```suggestion

* [Deploy an Ubuntu Azure VM and connect it to Azure Arc using ARM](azure_arc_servers_jumpstart/docs/azure_arm_linux.md)

* [Deploy a Windows Azure VM and connect it to Azure Arc using ARM](azure_arc_servers_jumpstart/docs/azure_arm_windows.md)
```

File: azure_arc_servers_jumpstart/azure/arm_template/azurevm_windows.json [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+      "defaultValue": "Standard_D2_v3",
+      "metadata": {
+        "description": "Size of the virtual machine."
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/azure/arm_template/azurevm_windows.json:73]:
```suggestion
        "description": "Azure region location for all resources."
```

File: azure_arc_servers_jumpstart/azure/arm_template/azurevm_windows.json [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+  "variables": {
+    "storageAccountName": "[concat('bootdiags', uniquestring(resourceGroup().id))]",
+    "publicIpName": "[concat(parameters('vmName'),'-PIP')]",
+    "nicName": "[concat(parameters('vmName'),'-NIC')]",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetName": "[concat(parameters('vmName'),'-SUBNET')]",
+    "subnetPrefix": "10.0.0.0/24",
+    "virtualNetworkName": "[concat(parameters('vmName'),'-VNET')]",
+    "subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnetName'))]",
+    "networkSecurityGroupName": "[concat(parameters('vmName'),'-NSG')]"
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/azure/arm_template/azurevm_windows.json:93]: This should also be the format and naming convention in the Linux json as well

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
+#  Onboard an Azure Linux Server VM with Azure Arc
+
+The following README will guide you on how onboard a Azure Linux VM on to Azure Arc. An ARM template is provided for the creation of the Azure Resouces, along with an script that will allow you to onboard the Azure VM onto Azure Arc, this step is requiered as Azure VMs are already part of ARM, therefore, the Azure Arc agent cannot be installed following the regular onboarding method. 
+
+   > [!NOTE]Please note that this scenario is only intended for demo purposes. 
+
+# Azure Account  
+
+* You will need an Azure Account with an active and valid subscription so you can deploy the Azure VMs and then register and onboard them with Azure Arc. If you do not have an account already, you can start with a free-trial account.
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/azure_arm_linux.md:9]: what does "valid" mean?

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
+#  Onboard an Azure Linux Server VM with Azure Arc
+
+The following README will guide you on how onboard a Azure Linux VM on to Azure Arc. An ARM template is provided for the creation of the Azure Resouces, along with an script that will allow you to onboard the Azure VM onto Azure Arc, this step is requiered as Azure VMs are already part of ARM, therefore, the Azure Arc agent cannot be installed following the regular onboarding method.
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/azure_arm_linux.md:3]:
```suggestion
The following README will guide you on how to onboard an Azure Linux VM on to Azure Arc. An ARM template is provided for the creation of the Azure Resouces, along with a script that will allow you to onboard the Azure VM onto Azure Arc, this step is required as Azure VMs are already part of ARM, therefore, the Azure Arc agent cannot be installed following the regular onboarding method. 
```

The onboarding to Arc should be automatic by default. We should also provide a "semi-automated" approach so the user can demo the onboarding itself by manually running the script

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md (hunk 11 above)
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/azure_arm_linux.md:9]:
```suggestion
* You will need an Azure Account with an active and valid subscription so you can deploy the Azure VMs and then register and onboard them with Azure Arc. If you do not have an account already, you can start with a free trial account. 
```

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+
+* Clone this repo
+
+    ```terminal
+    git clone https://github.com/microsoft/azure_arc.git
+    ```
+    
+* [Install or update Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest). **Azure CLI should be running version 2.7** or later. Use ```az --version``` to check your current installed version.
+
+# Automation Flow
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/azure_arm_linux.md:23]: Please align to the language [here](https://github.com/microsoft/azure_arc/blob/master/azure_arc_sqlsrv_jumpstart/docs/aws_terraform_winsrv.md#automation-flow)

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+
+Below you can find the automation flow for this scenario:
+
+1. Edit the *azurevm_linux.parameters.json* file 
+
+2. Upon execution of the *azurevm_linux.json* ARM template, an Azure VM will be created on your Azure Subscription
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/azure_arm_linux.md:29]:
```suggestion
2. Upon deployment of the *azurevm_linux.json* ARM template, an Azure VM will be created in your Azure Subscription
```

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+
+3. Connect to the VM and run the *allow_arc_agent.sh* Shell script on the guest OS to: 
+    * Prepare the OS for the installation of the "Azure Arc Connected Machine Agent".
+    * Install and configure the "Azure Arc Connected Machine Agent".
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/azure_arm_linux.md:33]: See my comment around automated and semi-automated deployment

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+    * Prepare the OS for the installation of the "Azure Arc Connected Machine Agent".
+    * Install and configure the "Azure Arc Connected Machine Agent". 
+
+# Deployment
+
+* Create a new Azure Resource Group where you want your machine(s) to be deployed and then be registered as Azure Arc enabled Servers. 
+
+    ```terminal
+    az group create --name <Name of the Azure Resource Group> --location <Azure Region>
+    ```
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/azure_arm_linux.md:41]: This should be automated as well

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+
+![](../img/azure_linux/01.png)
+
+* Before executing the ARM template, you must set the parameters that match your environment. Edit the *azurevm_linux.parameters.json* file and provide: 
+    - **adminUsername:** a username for Admin access to the Linux OS
+    - **adminPublicKey:** the public key for SSH access
+    - **dnsLabelPrefix:** a DNS prefix for the VM 
+    - **vmName:** a custom name for the Azure VM
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/azure_arm_linux.md:49]: This part should be the first part of the deployment phase (+ Resource Group)

File: azure_arc_servers_jumpstart/docs/azure_arm_windows.md [hunk 18]
```diff
@@ -0,0 +1,81 @@
+#  Onboard an Azure Windows Server VM with Azure Arc
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/azure_arm_windows.md:1]: Please review all my comments on the Linux guide and apply here

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md [hunk 19]
```diff
@@ -0,0 +1,86 @@
...
+
+# Deployment
+
+* Create a new Azure Resource Group where you want your machine(s) to be deployed and then be registered as Azure Arc enabled Servers. 
+
+    ```terminal
+    az group create --name <Name of the Azure Resource Group> --location <Azure Region>
+    ```
+
+![](../img/azure_linux/01.png)
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/azure_arm_linux.md:43]: Resource Group == Azure-Arc-Demo

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+![](../img/azure_linux/04.png)
+
+
+# Clean up environment
+
+Complete the following steps to clean up your environment.
+
+* Remove the resource group that holds all the resources for this scenario. 
+
+![](../img/azure_linux/05.png)
```
Comment (likamrat) [inline azure_arc_servers_jumpstart/docs/azure_arm_linux.md:86]: Fix Arc server name to align with the naming convention

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md (hunk 12 above)
Comment (lanicolas) [reply azure_arc_servers_jumpstart/docs/azure_arm_linux.md:3]: I did the onboarding by running a script, as automatic as possible. Were you thinking on a different way? Using custom scripts as far as the testing I did is not possible, happy to talk about this

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md (hunk 11 above)
Comment (lanicolas) [reply azure_arc_servers_jumpstart/docs/azure_arm_linux.md:9]: With credit basically. They cannot use a free account they had 5 years ago

File: azure_arc_servers_jumpstart/docs/azure_arm_linux.md (hunk 17 above)
Comment (lanicolas) [reply azure_arc_servers_jumpstart/docs/azure_arm_linux.md:49]: they are different scopes in ARM template deployments, this is at RG level the RG creation is at subscription.

**Pull request closed without merging**